Automattic/jetpack
# Protect: Turn on waf module by default and add UI control for automatic rules

#### Description
Adds to the Protect UI, the ability to toggle on automatic and manual rules independent of one another.

#### Changes proposed in this Pull Request:

* **The `waf` module is now enabled on plugin activation**.
* The firewall settings UI now expects to be able to use the `waf` package, and displays an error and prompt to re-enable the module if it is disabled (see screenshots).
* The module toggle in the firewall settings UI has been updated to control the `jetpack_waf_automatic_rules` option instead of toggling the module.

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1203279663034183

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:






- Checkout this branch
- Load your Jurassic Tube
- Validate that the `waf` module is disabled.
- Activate Protect
- Validate that the `waf` module is enabled.
- Proceed to Protect admin page and select the Firewall tab
- Verify that automatic and manual rules are disabled by default.
- Enable Automatic rules
- Validate that a request is blocked (i.e. `?foo=<script>`)
- Enable Manual rules
- Verify that a new manual rules settings section is loaded in place of the primary section
- Select Edit manual rules
- Add and/or remove rules and ensure that they are appropriately set
- Manually disable the `waf` module, and ensure the firewall settings screen matches the screenshot below.
- Ensure no regression in application functionality

#### Screenshots

<img width="1138" alt="Screen Shot 2022-12-12 at 11 26 45 AM" src="https://user-images.githubusercontent.com/10933065/207126307-9cad2646-696e-406b-867a-b92d22ed1896.png">

Module disabled:

<img width="1137" alt="Screen Shot 2022-12-12 at 11 35 38 AM" src="https://user-images.githubusercontent.com/10933065/207126642-8aaee49e-2fcc-4427-9d94-9349f3d2440e.png">
